### PR TITLE
Add more http proxy configs for native S3 filesystem implementation

### PR DESCRIPTION
**Pull request by anusudarsan**
To make the http proxy config options in parity with legacy implementation

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/object-storage/file-system-s3.md [hunk 1]
```diff
@@ -78,6 +78,15 @@ support:
 * - `s3.max-error-retries`
   - Specifies maximum number of retries the client will make on errors.
     Defaults to `10`.
+* - `s3.http-proxy.username`
```
Comment (nineinchnick) [inline docs/src/main/sphinx/object-storage/file-system-s3.md:81]: Move this higher, after existing `s3.http-proxy.*` options.

**Review by nineinchnick — APPROVED** [review]

File: lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java [hunk 2]
```diff
@@ -106,6 +106,10 @@ public S3FileSystemFactory(OpenTelemetry openTelemetry, S3FileSystemConfig confi
                     config.getHttpProxy()));
             httpClient.proxyConfiguration(ProxyConfiguration.builder()
                     .endpoint(endpoint)
+                    .username(config.getHttpProxyUsername())
```
Comment (findinpath) [inline lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java:109]: Curiosity question: Were you able to test these changes yourself?

Comment (anusudarsan) [reply lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java:109]: nope. just trusting the sdk :) seems straightforward

**Comment by mosabua** [comment]
This needs review from @electrum

**Review by electrum — APPROVED** [review]

**Pull request merged**